### ISSUE TITLE
No matching distribution found for dnet (from scapy-ssl_tls)

### ISSUE THREAD

**Issue by Chenneyfang** [tintinweb/scapy-ssl_tls#115]
i use "sudo pip install scapy-ssl_tls" to install,but it reported:
The directory '/Users/chenfangzheng/Library/Caches/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/Users/chenfangzheng/Library/Caches/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting scapy-ssl_tls
^Z
[3]  + 2675 suspended  sudo pip install scapy-ssl_tls
➜  ~ sudo -H pip install scapy-ssl_tls
Collecting scapy-ssl_tls
  Downloading scapy-ssl_tls-1.2.3.3.tar.gz (117kB)
    100% |████████████████████████████████| 122kB 29kB/s
Collecting pycrypto>=2.6 (from scapy-ssl_tls)
  Downloading pycrypto-2.6.1.tar.gz (446kB)
    100% |████████████████████████████████| 450kB 15kB/s
Collecting scapy<2.3.3,>=2.2.0 (from scapy-ssl_tls)
  Using cached scapy-2.3.2.tar.gz
Collecting tinyec>=0.3.1 (from scapy-ssl_tls)
  Using cached tinyec-0.3.1.tar.gz
Collecting dnet (from scapy-ssl_tls)
  Could not find a version that satisfies the requirement dnet (from scapy-ssl_tls) (from versions: )
No matching distribution found for dnet (from scapy-ssl_tls)

who can help me??

**Comment by tintinweb** [comment]
Hi @Chenneyfang,

afaik this is already fixed in master. Anyway, can you verify that #99 fixes the issue for you? 

change setup.py - swap`dnet` for `pydumbnet`. 
```
     if platform.system() == "Darwin":
 -        dependencies.extend(("dnet", "pcapy"))		 
+        dependencies.extend(("pydumbnet", "pcapy"))
```

**Comment by Chenneyfang** [comment]
yes,it is the fixed code,but i can't install it at the same time.

**Comment by Chenneyfang** [comment]
also reported:
>>> from scapy_ssl_tls.ssl_tls import TLS
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/scapy_ssl_tls/ssl_tls.py", line 691, in <module>
    class TLSHelloRetryRequest(Packet):
  File "/Library/Python/2.7/site-packages/scapy_ssl_tls/ssl_tls.py", line 695, in TLSHelloRetryRequest
    TypedPacketListField("extensions", None, TLSExtension, length_from=lambda x:x.length, type_="TLSHelloRetryRequest")]
  File "/Library/Python/2.7/site-packages/scapy_ssl_tls/ssl_tls.py", line 228, in __init__
    self.type_ = type_
AttributeError: 'TypedPacketListField' object has no attribute 'type_'

**Comment by Chenneyfang** [comment]
@tintinweb

**Comment by Chenneyfang** [comment]
[ssl_tls.txt](https://github.com/tintinweb/scapy-ssl_tls/files/1600667/ssl_tls.txt)
here is the code

**Comment by tintinweb** [comment]
Hi @Chenneyfang,

I've cherry-picked the fix from master to v1.2.3 and published it on pypi as [v1.2.3.4](https://github.com/tintinweb/scapy-ssl_tls/releases/tag/v1.2.3.4). That should fix your problem.

cheers,
tin
